src/service/link-rewriter/link-rewriter-manager.js

```diff
@@ -0,0 +1,231 @@
+   * LinkRewrited.id.
+   * @param {!../ampdoc-impl.AmpDoc} ampdoc
+   * @return {Array<string>}
+   */
+  getPriorityList_(ampdoc) {
+    const docInfo = Services.documentInfoForDoc(ampdoc);
+    const value = docInfo.metaTags[PRIORITY_META_TAG_NAME];
+
+    return value ? value.trim().split(/\s+/) : [];
+  }
+
+  /**
+   * @private
+   * Listen for DOM_UPDATE event.
+   * @param {*} iframeDoc
+   */
+  installGlobalEventListener_(iframeDoc) {
+    iframeDoc.addEventListener(AmpEvents.DOM_UPDATE,
+        this.onDomChanged_.bind(this));
+  }
+
+  /**
+   * @private
+   * Notify all the registered LinkRewriter when new elements have been added
+   * to the page so they can re-scan the page to find potential new links to
+   * replace.
+   */
+  onDomChanged_() {
+    this.linkRewriters_.forEach(linkRewriter => {
+      linkRewriter.onDomUpdated();
+    });
+  }
+
+
+  /**
+   * @private
+   * Extract the optional priority list for this specific anchor
+   * from its attribute. The 'data-link-rewriters' attribute should
+   * contain a whitespace separated list of LinkRewrited.id.
+   * @param {HTMLElement} anchor
+   * @return {Array<string>}
+   */
+  parseLinkRewriterPriorityForAnchor_(anchor) {
+    const dataValue = anchor.getAttribute('data-link-rewriters');
+    if (!anchor || !dataValue) {
+      return [];
+    }
+
+    return dataValue.trim().split(/\s+/);
+  }
+
+  /**
+   * @private
+   * Mutate linkRewriterList to insert a new LinkRewriter in the list
+   * while keeping the list in order of priority as defined by the
+   * idPriorityList.
+   * @param {Array<./link-rewriter.LinkRewriter>} linkRewriterList
+   * @param {!./link-rewriter.LinkRewriter} linkRewriter
+   * @param {Array<string>} idPriorityList
+   * @return {Array<./link-rewriter.LinkRewriter>} - Mutated linkRewriterList param.
+   */
+  insertInListBasedOnPriority_(linkRewriterList, linkRewriter, idPriorityList) {
+    const priorityIndex = idPriorityList.indexOf(linkRewriter.id);
+    if (priorityIndex > -1) {
+      linkRewriterList.splice(priorityIndex, 0, linkRewriter);
+    } else {
+      linkRewriterList.push(linkRewriter);
+    }
+
+    return linkRewriterList;
+  }
+
+  /**
+   * @private
+   * Get the list of all the link rewriter "watching" a specific anchor.
+   * See LinkRewriter.isWatchingLink for more details.
+   * @param {HTMLElement} anchor
+   * @return {Array<./link-rewriter.LinkRewriter>}
+   */
+  getSuitableLinkRewritersForLink_(anchor) {
+    const linkPriorityList = this.parseLinkRewriterPriorityForAnchor_(anchor);
+    return this.linkRewriters_.reduce((acc, linkRewriter) => {
+      if (linkRewriter.isWatchingLink(anchor)) {
+        this.insertInListBasedOnPriority_(acc, linkRewriter, linkPriorityList);
+      }
+
+      return acc;
+    }, []);
+  }
+}
+
+
+/**
+ * Register the global LinkRewriteService.
+ * @param {!../ampdoc-impl.AmpDoc} ampdoc
+ */
+export function installGlobalLinkRewriterServiceForDoc(ampdoc) {
```

Comment:
`registerGlobalLinkRewriterServiceForDoc` to stay consistent with the method description comment.

Reply:
`install` is the standard in the project, will rename the comment instead

Reply:
Actually I won't since it helps understanding what we mean by install :D